src/entrypoint/main.go

```diff
@@ -0,0 +1,69 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log"
+	"net/http"
+	"regexp"
+)
+
+type Response struct {
+	ProviderId          string `json:"providerId"`
+	ProviderName        string `json:"providerName"`
+	ProviderDisplayName string `json:"providerDisplayName"`
+	UrlSyncUX           string `json:"urlSyncUX"`
+	UrlAPI              string `json:"urlAPI"`
+	Width               int    `json:"width"`
+	Height              int    `json:"height"`
+}
+
+func main() {
+	http.HandleFunc("/", handleEntryPoint)
+
+	log.Fatal(http.ListenAndServe(":8080", nil))
+}
+
+func handleEntryPoint(w http.ResponseWriter, r *http.Request) {
+	err, host, port := parseUrl(r.URL.Path)
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	entryPointUrl := fmt.Sprintf(`https://%v:%v/modules/domain-connect/index.php/`, host, port)
+	response := Response{
+		ProviderId:          "plesk",
+		ProviderName:        "Plesk",
+		ProviderDisplayName: "Domain powered by Plesk",
+		UrlSyncUX:           entryPointUrl,
+		UrlAPI:              entryPointUrl,
+		Width:               750,
+		Height:              750,
+	}
+
+	w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+	w.WriteHeader(http.StatusOK)
+	if err := json.NewEncoder(w).Encode(response); err != nil {
+		panic(err)
+	}
+}
+
+func parseUrl(url string) (error, string, string) {
+	re := regexp.MustCompile(`(?:/host/(?P<host>.*?))?(?:/port/(?P<port>\d*?))?/v2/(?P<domain>.*?)/settings(/|$)`)
```

Comment:
Shouldn't compile regex on each request. You may either move it to local variable in function main, or as a package-level variable, initialized in function init()